app/todos/resources.py

```diff
@@ -0,0 +1,153 @@
+from flask import request, session
+from flask_restful import Resource
+from sqlalchemy import and_
+
+from app import db, Todo, TodoListPermission
+from app.lib import authentication, response_util, status
+
+
+class TodoMultiResource(Resource):
+    """Class for creating and accessing todos.
+    """
+    @authentication.requiresAuth
+    def get(self):
+        """Method gets a list of all the todos and returns them in an OK
+        response.
+        """
+        userId = session.get('userId')
+        permissions = TodoListPermission.query.filter_by(userId=userId)
```

Comment:
You should move this query into the if block below. Only when todoListId is None should we actually execute this query. DB queries are slow compared to other typical actions.